src/App.js

```diff
@@ -1,13 +1,96 @@
 import React from 'react';
+import TodoList from './components/TodoComponents/TodoList';
 
 class App extends React.Component {
   // you will need a place to store your state in this component.
   // design `App` to be the parent component of your application.
   // this component is going to take care of state, and any change handlers you need to work with your state
+  constructor() {
+    super();
+    this.state = {
+      userInput: '',
+      backgroundColors: ['#f1c40f', '#e67e22', '#e74c3c', '#1abc9c', '#2ecc71', '#3498db', '#9b59b6'],
+      todoList: [
+        {
+          task: 'Organize Garage',
+          id: 1528817077286,
+          completed: false,
+          backgroundColor: '#3498db'
+        },
+        {
+          task: 'Bake Cookies',
+          id: 1528817084358,
+          completed: false,
+          backgroundColor: '#9b59b6'
+        }
+      ]
+    };
+  }
+
+  enterPressed = event => {
+    if(event.which === 13) {
+      this.setState({
+        userInput: '',
+        todoList: this.state.todoList.concat({
+          task: event.target.value,
+          id: Date.now(),
+          completed: false,
+          backgroundColor: this.state.backgroundColors[Math.floor(Math.random()*this.state.backgroundColors.length)]
+        }) 
+      });
+    }
+  }
+
+  buttonPressed = () => {
+    this.setState({
+      userInput: '',
+      todoList: this.state.todoList.concat({
+        task: this.state.userInput,
+        id: Date.now(),
+        completed: false,
+        backgroundColor: this.state.backgroundColors[Math.floor(Math.random()*this.state.backgroundColors.length)]
+      }) 
+    });
+  }
+
+  handleSubmit = event => {
+    this.setState({
+      userInput: event.target.value
+    });
+  }
+
+  markDone = (id) => {
+    this.setState({
+      todoList: this.state.todoList.map(todo => {
```

Comment:
I would bring all your logic for updating the state out of the `setState` function. This will make your life easier when debugging. By following this pattern your calls to `setState` will look much simpler.